proxy/proxy/client.go

```diff
@@ -272,22 +274,32 @@ func (c *Client) startRefresh(instance string, refreshCfgBuffer time.Duration) c
 			logging.Errorf("failed to refresh the ephemeral certificate for %s, returning previous cert instead: %v", instance, err)
 			addr, cfg, ver, err = old.addr, old.cfg, old.version, old.err
 		}
-		c.cfgCache[instance] = cacheEntry{
+		e := cacheEntry{
 			lastRefreshed: time.Now(),
 			err:           err,
 			addr:          addr,
 			version:       ver,
 			cfg:           cfg,
 			done:          done,
 		}
+		c.cfgCache[instance] = e
 		c.cacheL.Unlock()
 
+		if !isValid(e) {
+			logging.Errorf("failed to refresh the ephemeral certificate for %v: %v", instance, err)
+			return
```

Comment:
We should probably add a note here about how this means future attempts won't be rescheduled unless another connection attempt is made